meson.build

```diff
@@ -0,0 +1,216 @@
+
+# Check for libsystemd availability
+want_systemd = get_option('systemd')
+if want_systemd != 'false'
+    libsystemd = dependency('libsystemd', required: want_systemd == 'true')
+    have = libsystemd.found()
+else
+    libsystemd = []
+    have = false
+endif
+conf.set('CONFIG_SYSTEMD', have, description: 'Is libsystemd required?')
+
+# Check for libjson-c availability
+want_json = get_option('json-c')
+if want_json != 'false'
+    libjson = dependency('json-c', required: false)
+    if not libjson.found()
+        libjson = cc.find_library('json-c', required: want_json == 'true')
+    endif
+
+    if not libjson.found() and want_json == 'true'
+        error('json-c is missing')
+    endif
+
+    have = libjson.found()
+else
+    libjson = []
+    have = false
+endif
+conf.set('CONFIG_JSONC', have, description: 'Is json-c required?')
+
+################################################################################
+# The following commented-out text is the beginning of an effort to replace
+# ccan/tools/configurator. It is not complete yet. Eventually we would like
+# to replace everything that ccan/tools/configurator does with meson.
+# To be continued...
+################################################################################
+#args = ['-g3', '-ggdb', '-Wall', '-Wundef', '-Wmissing-prototypes', '-Wmissing-declarations', '-Wstrict-prototypes', '-Wold-style-definition']
+#conf.set10(
+#    'HAVE_STRUCT_TIMESPEC',
+#    cc.compiles(
+#        '''
+#            #include <time.h>
+#            static void func(void) {
+#                struct timespec ts;
+#                ts.tv_sec = ts.tv_nsec = 1;
+#            }
+#            int main(int argc, char *argv[]) {
+#                (void)func();
+#                return 0;
+#            }
+#        ''',
+#        no_builtin_args: true,
+#        args: args,
+#        name: 'struct timespec'
+#    ),
+#    description: 'Is struct timespec defined?'
+#)
+#conf.set10(
+#    'HAVE_ASPRINTF',
+#    cc.compiles(
+#        '''
+#            #define _GNU_SOURCE
+#            #include <stdio.h>
+#            static char *func(int x) {
+#                char *p;
+#                if (asprintf(&p, "%u", x) == -1)
+#                    p = NULL;
+#                return p;
+#            }
+#            int main(int argc, char *argv[]) {
+#                (void)func(1000);
+#                return 0;
+#            }
+#        ''',
+#        no_builtin_args: true,
+#        args: args,
+#        name: 'asprintf()'
+#    ),
+#    description: 'Is asprintf() supported?'
+#)
+
+configure_file(
+    output: 'config-host.h',
+    configuration: conf
+)
+
+################################################################################
+NAME     = meson.project_name()
+SPECFILE = '@0@.spec'.format(NAME)
+
+substs = configuration_data()
+substs.set('NAME',       NAME)
+substs.set('VERSION',    run_command(awk, '/Version:/ { print $2 }', SPECFILE).stdout().strip())
```

Comment:
Could we get rid of awk here. Reading this from the specfile looks a bit awkward.

Reply:
Yes, we should avoid having two places defining the version. Hmm, not sure if we should remove the spec file.